docs/en/04.ui/07.tables.md

```diff
@@ -238,6 +238,33 @@ The query class must accept the query `Builder`, `Filter` instance, and is calle
 
 <div class="alert alert-primary">**Pro Tip:** Even automated stream filters can be completely overridden. Try providing your own custom query!</div>
 
+##### Dynamic Filter Options
+
+If you're creating a filter that uses a select box, your drop-down options can be either populated by a simple array, or dynamically. However to populate the fields dynamically, you'll need to create a constructor to perform your query as it's not currently possible to use a handler here.
+
+Your filter array would look something like this:
+
+    protected $filters = [
+        'status' => [
+            'placeholder' => 'Order Item Status',
+            'filter' => 'select',
+            'type' => 'select',
+            'query' => StatusFilterQuery::class,
+            'options' => [] // Note that this is intentionally empty.
+        ]
+    ];
+
+and your constructor would look like this:
+
+    public function __construct(Table $table, ItemStatusRepositoryInterface $status)
+    {
+        parent::__construct($table);
+
+        $this->filters['status']['options'] = $status->all()->pluck('name', 'id')->all();
```

Comment:
This is bullshit! You don't need to do it. Creating such of a constructor is the bad practice. You need to create a new class for your filter, which should extends one of filter classes of platform.

Also `$this->filters['status']['options'] =` is the very-very bad practice too!!!

Reply:
@rickmills first let me apologize on behalf of the project. This behavior is 100% not acceptable behavior and I appreciate you helping out! I've taken appropriate measures to ensure this kind of attitude is no longer an issue from him as well.


That said - dynamic options in a select filter can be handled in a couple ways:

- You can use a `Filters` handler and set them there with a closure when you set the filters.

- You can/should be able to use a handler for options in the filter:

```
    'status' => [
            'filter'  => 'select',
            'options' => 'Example\TestClass@handle',
        ],
```

I REALLY appreciate you digging into the docs. It's a sore spot for us and the help is huge! Take a look at these approaches and let me know what you think about them. I spot tested the handler and it's working so you might have had something off. Check it out though!

I also apologize for not getting to this sooner, for multiple reasons. I was out of town for a week and have been catching up and reorganizing efforts.